File: nowcoder/question/NC52.cpp
class Solution {
public:
    /**
     *
     * @param s string×Ö·û´®
     * @return bool²¼¶ûÐÍ
     */
    bool isValid(string s) {
        // write code here
        stack<char> st;
        if (s.length() % 2 == 1) {
            return false;
        }
        for (int i = 0; i < s.length(); ++i) {
            if (s[i] == '(' || s[i] == '[' || s[i] == '{') {
                st.push(s[i]);
            }
            else {
                if (st.empty()) {
                    return false;
                }
                if (st.top() == '(' && s[i] == ')') {
                    st.pop();
                }
                else if (st.top() == '[' && s[i] == ']') {
                    st.pop();
                }
                else if (st.top() == '{' && s[i] == '}') {
                    st.pop();
                }
            }
        }
        if (st.empty()) {
            return true;
        }
        return false;
    }
};

File: nowcoder/question/NC52.py
class Solution:
    def isValid(self , s ):
        # write code here
        stack = []
        mp = {')':'(', ']':'[', '}':'{'}
        for i in range(len(s)):
            if not(s[i] in mp):
                stack.append(s[i])
            elif (len(stack) == 0) or (stack[len(stack) - 1] != mp[s[i]]):
                return False
            else:
                stack.pop()
        if(len(stack) == 0):
            return True
        else:
            return False